src/Language/PureScript/TypeChecker.hs

```diff
@@ -280,12 +293,16 @@ typeCheckAll moduleName _ = traverse go
     addTypeClass moduleName pn args implies deps tys
     return d
   go (d@(TypeInstanceDeclaration dictName deps className tys body)) = rethrow (addHint (ErrorInInstance className tys)) $ do
-    traverse_ (checkTypeClassInstance moduleName) tys
-    checkOrphanInstance dictName className tys
-    _ <- traverseTypeInstanceBody checkInstanceMembers body
-    let dict = TypeClassDictionaryInScope (Qualified (Just moduleName) dictName) [] className tys (Just deps)
-    addTypeClassDictionaries (Just moduleName) . M.singleton className $ M.singleton (tcdValue dict) dict
-    return d
+    env <- getEnv
+    case M.lookup className (typeClasses env) of
+      Nothing -> internalError "typeCheckAll: Encountered unknown type class in instance declaration"
```

Comment:
Am I correct in the assumption that by this point the referenced type class should already have been added to the environment?